javascript/javascript2/week3/3the_fastest_presser_in_this_realm/index.js

```diff
@@ -0,0 +1,64 @@
+
+
+//const time = document.getElementById("seconds");
+
+//getting the value of an input
+function onClick(elem) {
+  const parent = elem.parentNode;
+  const value = parent.querySelector('input[type=text]').value;
+  if (value == '') {
+    console.log('no input');
+  } else {
+    console.log(value);
+  }
+ /*because I need to access the value variable, and I can't do that outside this function because of the scope, I will keep adding timeout function inside this function */
+
+  /*the next 2 functions are taking the value(input by user, say 3 or 5, or other) and multiplies it by 1000 because timeout function will work with seconds in the format 1000 for 1 second*/
+  function multiply(num1, num2){
+    return num1 * num2;
+  }
+  //creating result variable before so that I can access it after
+ let result;
+function myValue (num1, num2, operator) {
+ result = operator(num1, num2);
+  return result
+}
+  myValue(value, 1000, multiply);
+
+
+//setting timeout
+const myTimeout = setTimeout(myString, result);
+
+function myString() {
+  document.getElementById("hidden").innerHTML = "Time's up!"
+var confettiSettings = { target: 'my-canvas' };
+var confetti = new ConfettiGenerator(confettiSettings);
+confetti.render();
+}
+  };
+
+//a function when any key is pressed
+document.getElementById("input1").addEventListener("keydown", keyPressed);
+
+function keyPressed() {
+  document.getElementById("hidden").innerHTML = "You pressed a key!";
+};
+document.addEventListener("keypress", function(event) {
+  //whichKey(event.key);
```

Comment:
```
`const inputTimer = document.getElementById("inputTimer");
const startBtn = document.getElementById("btnGameStrt");
const player1 = document.getElementById("userName1");
const player2 = document.getElementById("userName2");
const count1 = document.getElementById("count1");
const count2 = document.getElementById("count2");

let player1PressCount = 0;
let player2PressCount = 0;

startBtn.addEventListener("click", gameStart);

function gameStart() {
    inputTimer.disabled = true;
    startBtn.disabled = true;
    const totlaNumberOfSeconds = inputTimer.value;
    // Add an event listener for key presses
    document.addEventListener('keyup', countKeyPress)
    setTimeout(endGame, totlaNumberOfSeconds * 1000)

}
function endGame() {
    document.removeEventListener('keyup', countKeyPress)
    if (player1PressCount > player2PressCount) {
        player1.innerHTML += "<br> winner";
        player2.innerHTML += "";

    }
    else if (player1PressCount < player2PressCount) {
        player1.innerHTML += "";
        player2.innerHTML += "<br> winner";

    } else {
        // if both players have same press count game is tie and game over
        player1.innerHTML += " <br> draw";
        player2.innerHTML += " <br> draw";

    }
}

function countKeyPress(e) {
    if (inputTimer.value === 0) {
        e.preventDefault()
        return
    }
    if (e.keyCode === 83) {
        player1PressCount++;
        count1.innerHTML = player1PressCount;
    }

    if (e.keyCode === 76) {
        player2PressCount++
        count2.innerHTML = player2PressCount;
    }

}

`
```
you can take a look on this script it may help you